config/routes.rb

```diff
@@ -123,7 +123,12 @@
     resource :search,    only: :show
     resource :dashboard, only: :show, constraints: { format: /html|atom/ }
     resources :profiles, only: :show
-    resource :profile, only: [:edit, :update]
+    resource :profile, only: [:edit, :update] do
+      member do
+        get :delete
+        delete :destroy, as: :destroy
```

Comment:
I may be failing to see the appeal of a separate controller. Can someone please explain? `ProfilesController` handles `edit` and `update` of profiles, I would believe that `delete` and `destroy` are related concepts. The controller isn't large either.
Currently routes look like:
```Ruby
delete_profile    GET     /profile/delete(.:format)    profiles#delete {:format=>:html}
destroy_profile   DELETE  /profile(.:format)           profiles#destroy {:format=>:html}
edit_profile      GET     /profile/edit(.:format)      profiles#edit {:format=>:html}
                  PATCH   /profile(.:format)           profiles#update
```
I am sorry but how is this non intuitive?